guvnor-services/guvnor-services-backend/src/test/java/org/guvnor/common/services/backend/file/upload/AbstractFileServletTest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.common.services.backend.file.upload;
+
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+import org.jboss.errai.bus.client.api.QueueSession;
+import org.jboss.errai.bus.server.api.SessionProvider;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.uberfire.backend.vfs.Path;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AbstractFileServletTest {
+
+    @Mock
+    private SessionProvider provider;
+
+    @Mock
+    private HttpServletRequest request;
+
+    @Mock
+    private QueueSession queueSession;
+
+    @Mock
+    private HttpSession httpSession;
+
+    @Test
+    public void testGetSessionId() throws Exception {
```

Comment:
@karreiro You know what I'm going to ask, don't you?..... How about a test for when the request parameter is missing?